synthesis/codon/analyzer_test.go

```diff
@@ -0,0 +1,57 @@
+package codon
+
+import (
+	"testing"
+
+	"github.com/TimothyStiles/poly/io/genbank"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestComputeStats(t *testing.T) {
+	t.Parallel()
+
+	sortStrings := cmpopts.SortSlices((func(a, b string) bool { return a < b }))
+
+	tests := []struct {
+		name string
+
+		testDataPath string
+		wantStats    *Stats
+
+		wantErr error
+	}{
+		{
+			name: "pUC19 plasmid",
+
+			testDataPath: "../../data/puc19.gbk",
+
+			wantStats: &Stats{
+				StartCodonCount: map[string]int{"atg": 2},
+			},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		var tt = tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			sequence, err := genbank.Read(tt.testDataPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			analyzer := &Analyzer{}
+
+			stats, err := analyzer.ComputeStats(sequence)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !cmp.Equal(stats, tt.wantStats, sortStrings) {
+				t.Errorf("stats and tt.wantStats didn't match %s", cmp.Diff(stats, tt.wantStats, sortStrings))
+			}
```

Comment:
No need to sort strings. `cmp.Diff()` compares by key. Would recommend the following instead:

```suggestion
			if diff := cmp.Diff(stats, tt.wantStats); diff != "" {
				t.Errorf("stats and tt.wantStats didn't match (-got,+want): %s", diff)
			}
```

Reply:
ah yea, I always get confused with cmp ordering rules...